runtime/debugger.c

```diff
@@ -109,24 +109,23 @@ static void open_connection(void)
   SOCKET sock = WSASocket(sock_domain, SOCK_STREAM, 0,
                           NULL, 0,
                           0 /* not WSA_FLAG_OVERLAPPED */);
-  if (sock != INVALID_SOCKET &&
-      connect(sock, &sock_addr.s_gen, sock_addr_len) == 0)
-    dbg_socket = _open_osfhandle(sock, 0);
-  if (dbg_socket == -1) {
+  if (sock == INVALID_SOCKET
+      || connect(sock, &sock_addr.s_gen, sock_addr_len) != 0)
+    caml_fatal_error("cannot connect to debugger at %s\n"
+                     "WSA error code: %d",
+                     (dbg_addr ? dbg_addr : "(none)"),
+                     WSAGetLastError());
+  dbg_socket = _open_osfhandle(sock, 0);
+  if (dbg_socket != -1)
```

Comment:
Shouldn't this be `== -1`? We're in the error path of `_open_osfhandle` and we want to fall into the shared  `caml_fatal_error`.

Reply:
You're right, apologies for my mistake.  I just pushed a fix.